# MANOVA

Manova is a multivariate extension of ANOVA. It is usedot model tow or more dependent variables that are continuous with one or more categorical predictor ariables. It is often used to assess for differences between two or more groups.


To perform Multivariate Analysis  of Variace (MANOVA) in Python, we typically use the `statesmodels` library. 
Manova is used when there are two or more dependent varibles and one or more independent varibles. It takes whether the mean differences among groups on a combiantion fo dependent varibles are likely to have occurred by chance. 

Here's an example demonstration haw to create a MANOVA table in Python.


## Example: MANOVA with StatsModels 

Let's say we have a dataset with two dependetn varibles (e.g., test score in mathematics ans science) and one independet varible (e.g., teacing method). We want to now if there are statistically significatn differences in the dependent varibales across the levels of the independt variable. 


## Explanation:

- **Dataset Preparation:** The `data` dictionary and `DataFrame (df)` contain the sample data. Replace this with your actual data.
- **MANOVA Execution:** The `Manova. form_formula` method is used to peroform the MANOVA. The formula 'MathScore+ScienceScore~Method' indicates that `MathScore` and `ScienceScore ` are depenetn variables, and `Method` si the independent variable.
- **Result:** The `mv_test()` mehtod is used to get the MANOVA test results, which sar printed to the console.


This script will optput the MANOVA able, including Pillai's trace, Wilks' lambda, Hotelling-Lawley trace, and Roy's greatest root test statistics, along with their associated F-values, degrees of freedom and p-values, These reults will help you deerine if there are satisically significant diffeences in the dependent variables across the levels of the independent variable.



In [4]:
import pandas as pd 
import statsmodels.api as sm 
from statsmodels.multivariate.manova import MANOVA

# Example dataset

data ={
    'Method':['A','A','A','B','B','B','C','C','C'],
    'MathScore':[20,22,21,19,18,20,22,23,21],
    'ScienceScore':[30,28,29,33,32,31,29,27,28]
}

df = pd.DataFrame(data)
print(df.head(10))

#Perform MANOVA
maov = MANOVA.from_formula("MathScore + ScienceScore ~ Method", data=df)
print(maov.mv_test())


  Method  MathScore  ScienceScore
0      A         20            30
1      A         22            28
2      A         21            29
3      B         19            33
4      B         18            32
5      B         20            31
6      C         22            29
7      C         23            27
8      C         21            28
                   Multivariate linear model
                                                                
----------------------------------------------------------------
       Intercept          Value   Num DF Den DF  F Value  Pr > F
----------------------------------------------------------------
          Wilks' lambda    0.0005 2.0000 5.0000 4711.5000 0.0000
         Pillai's trace    0.9995 2.0000 5.0000 4711.5000 0.0000
 Hotelling-Lawley trace 1884.6000 2.0000 5.0000 4711.5000 0.0000
    Roy's greatest root 1884.6000 2.0000 5.0000 4711.5000 0.0000
----------------------------------------------------------------
                              

## Interpertation of MANOVA Results

The MANOVA results provided contain two main parts: The test statistics associated with the independent variable (Method). Each part includes four part includes four differnet test statistics: Wilks' lambda, Pillai's trace, Hotelling-Lawley trace, and Roy's greatest root. Let's interpret these result:


### Intercept Part 
1. **Wilks' Lambda:** A value close to 0 (0.0005) with a significant F-value (4711.5) and a p-value of 0.0000 indicates that the model with the intercept is significantly different from a model without the intercept.
2. **Pillai's Trace:** Similar to Wilks' lambda, a value close to 1 (0.9995) with a significant F-value and p-value indicates strong model significance.
3. **Hotelling-Lawley Trace:** A value of 4.4 with an F-value of 5.4 and a p-value of 0.0444. This p-value is below 0.05, indicating significant differences in the group means.
4. **Roy's Greatest Root:** A value of 4.3656, with an F-value of 13.0969 and a p-value of 0.0065. This result suggests significant differences in the group means, as indicated by this low p-value.


## Overall Interpretation

- Teh significatn intercept part indicates that the overall models is significant. 
- For the `Method` pat, different test statisteic provide somewhat conflictiong result. Wilks' Lambda and Pillai's Treace suggest aht the means of different methods are not significantly different, while Hotelling-Lawley Trace ans Roy's Greatest Root suggest singificant diffeences.
- Such discrepancies can occur due to the sensitivity of each test to diffeent assumptions and ata characteristeis, in parctice, when result confkict, it's often advisable to futher invisable to futher investigate the data , potentially considering other forms fo analysis or lookin into specific pairwise comparisons for more insights.
______________________________________________________________________